tests/linalg_test.py

```diff
@@ -1130,9 +1128,7 @@ def testSolveBroadcasting(self, lhs_shape, rhs_shape):
     args_maker = lambda: [rng(lhs_shape, 'float32'), rng(rhs_shape, 'float32')]
 
     if jtu.numpy_version() >= (2, 0, 0):
-      # TODO(jakevdp) remove this condition after solve broadcast deprecation is finalized.
-      if len(rhs_shape) == 1 or (len(lhs_shape) != len(rhs_shape) + 1):
-        self._CheckAgainstNumpy(np.linalg.solve, jnp.linalg.solve, args_maker, tol=1E-3)
+      self._CheckAgainstNumpy(np.linalg.solve, jnp.linalg.solve, args_maker, tol=1E-3)
```

Comment:
This test seems to be failing the "oldest supported numpy" workflow. Does this need a condition on numpy version?

Reply:
I removed the check for older NumPy versions – that's probably the cleanest approach here, since the semantics of batched right-hand values changed in NumPy 2.0